xml/depl_maintenance.xml

```diff
@@ -1503,6 +1503,295 @@ migrate</screen>
     the selected nodes.
    </para>
   </sect2>
+  <sect2 xml:id="sec.depl.maintenance.upgrade.troubleshooting">
+   <title>Common Upgrade Problems and Possible Solutions</title>
+    <qandaset>
+     <qandaentry>
+      <question>
+       <para>
+	Upgrade of the admin server has failed.
+       </para>
+      </question>
+      <answer>
+       <para>
+	Check for empty, broken, and not signed repositories in the &admserv;
+	upgrade log file <filename>/var/log/crowbar/admin-server-upgrade.log</filename>. Fix the
+	repository setup. Upgrade then remaining packages manually to &cloudos;
+	and &cloud; &productnumber; using the command <command>zypper dup</command>. Reboot the &admserv;.
+       </para>
+      </answer>
+     </qandaentry>
+     <qandaentry>
+      <question>
+       <para>
+	Upgrade step repeatedly fails due to timeout.
+       </para>
+      </question>
+      <answer>
+       <para>
+	Timeouts for most upgrade operations can be adjusted in the
+	<filename>/etc/crowbar/upgrade_timeouts.yml</filename> file. If the
+	file doesn't exist, use the following template, and modify it to your needs:
+       </para>
+       <screen>
+	:prepare_repositories: 120
+        :pre_upgrade: 300
+        :upgrade_os: 1500
+        :post_upgrade: 600
+        :shutdown_services: 600
+        :shutdown_remaining_services: 600
+        :evacuate_host: 300
+        :chef_upgraded: 1200
+        :router_migration: 600
+        :lbaas_evacuation: 600
+        :set_network_agents_state: 300
+        :delete_pacemaker_resources: 600
+        :delete_cinder_services: 300
+        :delete_nova_services: 300
+        :wait_until_compute_started: 60
+        :reload_nova_services: 120
+        :online_migrations: 1800
+       </screen>
+       <para>
+	The following entries may require higher values:
+       </para>
+	<itemizedlist>
+	 <listitem>
+	  <para>
+	   <literal>upgrade_os</literal> Time allowed for upgrading all packages of one node.
+	  </para>
+	 </listitem>
+	 <listitem>
+	  <para>
+	   <literal>chef_upgraded</literal> Time allowed for initial crowbar_join and chef-client run on a node that has been upgraded and rebooted.
+	  </para>
+	 </listitem>
+	 <listitem>
+	  <para>
+	   <literal>evacuate_host</literal> Time allowed for live migrate all VMs from a host.
+	  </para>
+	 </listitem>
+	</itemizedlist>
+      </answer>
+     </qandaentry>
+     <qandaentry>
+      <question>
+       <para>
+	Node upgrade has failed during live migration.
+       </para>
+      </question>
+      <answer>
+       <para>
+	The problem may occur when it is not possible to live
+	migrate certain VMs anywhere. It may be necessary to shut down or suspend
+	other VMs to make room for migration. Note that the Bash shell script
+	that starts the live migration for the &compnode; N is executed from
+	the &contrnode; M. Check the
```

Comment:
We can probably leave those *N* and *M*'s as we're not using them in the text anyway.

Reply:
And we should mention that both nodes are mentioned in the error message, something like, "Note that the Bash shell script that starts the live migration for the &compnode; is executed from the &contrnode; names of both nodes are mentioned in the error message that comes from 'crowbarctl upgrade status'"